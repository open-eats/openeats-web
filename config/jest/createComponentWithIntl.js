import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

// See: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;