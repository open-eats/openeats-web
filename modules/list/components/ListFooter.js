import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { injectIntl, defineMessages } from 'react-intl'

import { 
  ALL_ITEMS, 
  ACTIVE_ITEMS, 
  COMPLETED_ITEMS 
} from '../constants/ListStatus'

const ListFooter = ({activeFilter, completedCount, itemCount, onClearCompleted, onFilterStatus, intl }) => {
  const { formatMessage } = intl;
  const messages = defineMessages({
    itemsLeft: {
      id: 'list.footer.items_left',
      description: 'Number of items left',
      defaultMessage: '{itemCount, plural, =0 {No items} one {1 item left} other {{itemCount} items left}}',
    },
    all: {
      id: 'list.footer.all',
      description: 'Show all items',
      defaultMessage: 'All',
    },
    completed: {
      id: 'list.footer.completed',
      description: 'Show only completed items',
      defaultMessage: 'Completed',
    },
    active: {
      id: 'list.footer.active',
      description: 'Show active items',
      defaultMessage: 'Active',
    },
    clearCompleted: {
      id: 'list.footer.clear_completed',
      description: 'Clear all completed list items',
      defaultMessage: 'Clear completed',
    }
  });

  let clearButton = null;

  if (completedCount > 0) {
    clearButton = (
      <button
        className="clear-completed clear-button"
        onClick={ onClearCompleted }>
        { formatMessage(messages.clearCompleted) }
      </button>
    );
  }

  return (
    <div className="list-footer">
      <span className="list-count">
        { formatMessage(messages.itemsLeft, {itemCount: itemCount}) }
      </span>
      <ul className="filters">
        <li>
          <p
            className={ classNames({ selected: activeFilter === ALL_ITEMS })}
            onClick={ () => { onFilterStatus(ALL_ITEMS) }}>
              { formatMessage(messages.all) }
          </p>
        </li>
        { ' ' }
        <li>
          <p
            className={ classNames({ selected: activeFilter === ACTIVE_ITEMS })}
            onClick={ () => { onFilterStatus(ACTIVE_ITEMS) }}>
              { formatMessage(messages.active) }
          </p>
        </li>
        { ' ' }
        <li>
          <p
            className={ classNames({ selected: activeFilter === COMPLETED_ITEMS })}
            onClick={ () => { onFilterStatus(COMPLETED_ITEMS) }}>
              { formatMessage(messages.completed) }
          </p>
        </li>
      </ul>
      { clearButton }
    </div>
  );
};

ListFooter.propTypes = {
  itemCount: PropTypes.number.isRequired,
  completedCount: PropTypes.number.isRequired,
  activeFilter: PropTypes.string.isRequired,
  onClearCompleted: PropTypes.func.isRequired,
  onFilterStatus: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
};

export default injectIntl(ListFooter)
