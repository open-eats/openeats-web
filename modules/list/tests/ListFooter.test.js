import React from 'react';
import ListFooter from '../components/ListFooter';
import createComponentWithIntl from '../../../config/jest/createComponentWithIntl';

import {
  ALL_ITEMS,
  ACTIVE_ITEMS,
  COMPLETED_ITEMS
} from '../constants/ListStatus'

test('MyLists component test', () => {
  const mockClearCompleted = jest.fn();
  const mockFilterStatus = jest.fn();
  const component = createComponentWithIntl(
    <ListFooter
        activeFilter={ ALL_ITEMS }
        completedCount={ 3 }
        itemCount={ 6 }
        onClearCompleted={ mockClearCompleted }
        onFilterStatus={ mockFilterStatus }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MyLists component test w/ active filter', () => {
  const mockClearCompleted = jest.fn();
  const mockFilterStatus = jest.fn();
  const component = createComponentWithIntl(
    <ListFooter
        activeFilter={ ACTIVE_ITEMS }
        completedCount={ 3 }
        itemCount={ 6 }
        onClearCompleted={ mockClearCompleted }
        onFilterStatus={ mockFilterStatus }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MyLists component test w/ completed filter', () => {
  const mockClearCompleted = jest.fn();
  const mockFilterStatus = jest.fn();
  const component = createComponentWithIntl(
    <ListFooter
        activeFilter={ COMPLETED_ITEMS }
        completedCount={ 3 }
        itemCount={ 6 }
        onClearCompleted={ mockClearCompleted }
        onFilterStatus={ mockFilterStatus }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
