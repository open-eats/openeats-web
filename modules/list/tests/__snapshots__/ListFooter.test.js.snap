// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLists component test 1`] = `
<div
  className="list-footer"
>
  <span
    className="list-count"
  >
    6 items left
  </span>
  <ul
    className="filters"
  >
    <li>
      <p
        className="selected"
        onClick={[Function]}
      >
        All
      </p>
    </li>
     
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        Active
      </p>
    </li>
     
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        Completed
      </p>
    </li>
  </ul>
  <button
    className="clear-completed clear-button"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;

exports[`MyLists component test w/ active filter 1`] = `
<div
  className="list-footer"
>
  <span
    className="list-count"
  >
    6 items left
  </span>
  <ul
    className="filters"
  >
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        All
      </p>
    </li>
     
    <li>
      <p
        className="selected"
        onClick={[Function]}
      >
        Active
      </p>
    </li>
     
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        Completed
      </p>
    </li>
  </ul>
  <button
    className="clear-completed clear-button"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;

exports[`MyLists component test w/ completed filter 1`] = `
<div
  className="list-footer"
>
  <span
    className="list-count"
  >
    6 items left
  </span>
  <ul
    className="filters"
  >
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        All
      </p>
    </li>
     
    <li>
      <p
        className=""
        onClick={[Function]}
      >
        Active
      </p>
    </li>
     
    <li>
      <p
        className="selected"
        onClick={[Function]}
      >
        Completed
      </p>
    </li>
  </ul>
  <button
    className="clear-completed clear-button"
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;
